core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -944,6 +944,10 @@ public XContentBuilder dateValueField(XContentBuilderString rawFieldName, XConte
         return this;
     }
 
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, TimeValue timeValue) throws IOException {
```

Comment:
This seems weird, because if someone does:

```java
timeValueField("took_in_nanos", "took", TimeValue.timeValueNanos(1234L))
```

They'll get a field called "took_in_nanos" that's actually milliseconds!